File: Daily/sbgo/Leetcode80-删除排序数组中的重复项II.go
package sbgo

func removeDuplicates(nums []int) int {
	ptr, count := 1, 1
	for i := 1; i < len(nums); i++ {
		if nums[i] == nums[i-1] {
			count++
		} else {
			count = 1
		}

		if count <= 2 {
			nums[ptr] = nums[i]
			ptr++
		}
	}
	return ptr
}

File: Daily/PY/Leetcode80-删除排序数组中的重复项II.py
from typing import List
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        count,ptr = 1,1
        n = len(nums)
        for i in range(1,n):
            if nums[i-1] == nums[i]:
                count += 1
            else:
                count = 1
            
            if count <= 2:
                nums[ptr] = nums[i]
                ptr += 1

        return ptr
